guides/v2.1/cloud/cdn/fastly-vcl-badreferer.md

```diff
@@ -11,57 +11,104 @@ functional_areas:
-6. Select the checkbox for **Activate after the change** if you want to the dictionary after creating or editing the container.
-7. Add key-value pairs in the new dictionary. For this example, enter the URLs for your blog that should be redirected to your Wordpress backend. Enter a value of 1.
+
+## Create a referrer block list
+
+Edge Dictionaries create key-value pairs accessible to VCL functions during VCL snippet processing. In this example, you create an edge dictionary that provides the list of website referrers that you want to block.
+
+1.  Log in to the Admin UI for your {{ site.data.var.ece }} project environment.
+
+1.  Navigate to **Stores** > **Configuration** > **Advanced** > **System** > **Fastly Configuration**.
+
+1.  Expand the **Edge dictionaries** section.
+
+1.  Create the Dictionary container:
+
+    - Click **Add container**.
+
+    -  On the *Container* page, enter a **Dictionary name**. For this example, use the name `referrer_blocklist`.
+
+    -  Select **Activate after the change** to deploy your changes to the service version that you are editing.
+
+    -  Click **Upload** to attach the dictionary to your Fastly service configuration.
+
+1.  Add the list of domain names to block to the the `referrer_blocklist` dictionary:
+
+    -  Click the Settings icon for the `referrer_blocklist` dictonary.
+
+    -  Add and save key-value pairs in the new dictionary. For this example, each **Key** is a website referrer URL that you want to block and the **Value** is `true`. 
+       
+       ![Add bad referrer dictionary items]
+	 
+    -  Click **Cancel** to return to the system configuration page.
 
 For more information on using Edge Dictionaries with your VCL snippets, see Fastly's [Creating and using Edge Dictionaries](https://docs.fastly.com/guides/edge-dictionaries/creating-and-using-dictionaries) and their example [custom VCL snippets](https://docs.fastly.com/guides/edge-dictionaries/creating-and-using-dictionaries#custom-vcl-examples).
 
-## Create badreferer.json {#vcl}
+## Create a custom VCL snippet to block referrer spam
 
-For this example, you may only want to run it against the Production server. You can also add it to Staging for testing.
+The following custom VCL snippet code (JSON format) checks if incoming requests are from a referrer site included in the `referrer_blocklist` edge dictionary. If the site is in the block list, the request is blocked.
 
-Create an `badreferer.json` file with the following JSON content:
 
 ```json
 {
-  "name": "badreferer",
+  "name": "block_bad_referrer",
   "dynamic": "0",
   "type": "recv",
   "priority": "5",
   "content": "set req.http.Referer-Host = regsub(req.http.Referer, \"^https?://?([^:/\\s]+).*$\", \"\\1\"); if (table.lookup(referer_blocklist, req.http.Referer-Host)) { error 403 \"Forbidden\"; }"
 }
 ```
+Review the code to determine if you need to change any values:
+
+  -  `name`: Name for the VCL snippet. For this example, we used the name `block_bad_referrer`.
```

Comment:
This list format is difficult to read.
Consider using:

-  `name`—Name for the VCL snippet.

Or use a table? That might be too big.